bbs/input.h

```diff
@@ -54,7 +54,7 @@ std::string input_password(const std::string& prompt_text, int max_length);
 template<typename T>
 typename std::enable_if<std::is_signed<T>::value, T>::type
 input_number(T current_value, T min_value, T max_value, bool set_default_value = true) {
-  int len = std::max<int>(1, static_cast<T>(ceil(log10(max_value))));
+  int len = std::max<int>(1, static_cast<T>(ceil(log10((double)max_value))));
```

Comment:
@apamment what compiler are you using? C++11 has a log10 that takes T for the type passed.   Does log10<T>(max_value) compile?